### PR TITLE
Add Login Modal on "Join Public Lab" link

### LINKED ISSUE #4193: Login modal on join public lab link
This issue is a continuation of #4132 

On the https://publiclab.org/research page, there is a link that says "Welcome! This could be your dashboard if you joined Public Lab"
image

What we want to do is to make a login modal appear when that link has been clicked.

### PR DESCRIPTION
**Pull request by dinaelhanan**
Fixes  #4193 (<=== Add issue number here)

* [x] tests pass -- look for a green checkbox ✔️ a few minutes after opening your PR -- or run tests locally with `rake test`
* [x] code is in uniquely-named feature branch and has no merge conflicts
* [x] PR is descriptively titled
* [x] ask `@publiclab/reviewers` for help, in a comment below

### CODE REVIEW

**Comment by dinaelhanan** [comment]
@publiclab/reviewers check it out please! @SidharthBansal

File: test/fixtures/node_tags.yml [hunk 1]
```diff
@@ -9,7 +9,7 @@ awesome:
   uid: 2
```
Comment (SidharthBansal) [inline test/fixtures/node_tags.yml]: Please remove these changes.

**Comment by dinaelhanan** [comment]
@SidharthBansal i'm having issues when I preview this on cloud 9:
vlcrecord2018120912h07m47s

File: app/views/dashboard/_header.html.erb [hunk 2]
```diff
@@ -36,7 +36,7 @@
       <%= t('dashboard._header.you_shared') %> <a href="/profile/<%= current_user.username %>"><%= @user_note_count %> <% if @user_note_count <= 1 %><%= t('dashboard._header.research_note') %><% else %><%= t('dashboard._header.research_notes') %><% end %></a>.
     <% else %>
       <span class="hidden-xs hidden-sm"><%= t('dashboard._header.welcome') %>!</span>
-      <%= t('dashboard._header.this_could_be_dashboard') %> <a href="/signup"><%= t('dashboard._header.joined_public_lab') %></a>
+      <%= t('dashboard._header.this_could_be_dashboard') %> <a data-toggle="modal" data-target="#loginModal"><%= t('dashboard._header.joined_public_lab') %></a>
```
Comment (SidharthBansal) [inline app/views/dashboard/_header.html.erb]: Changes looks good to me!!
Great!!!

**Comment by SidharthBansal** [comment]
ok. Is the login modal working on the cloud9 platform?

**Comment by dinaelhanan** [comment]
@SidharthBansal how do I check?

**Comment by SidharthBansal** [comment]
By clicking on the header login button.

**Comment by dinaelhanan** [comment]
@SidharthBansal no.. I am getting the old login page. Also, how do I undo that commit? I am trying revert but it is not working

**Comment by SidharthBansal** [comment]
You are working on previous state of master. Please do a rebase.

**Comment by dinaelhanan** [comment]
@SidharthBansal it says everything is up to date:
screenshot from 2018-12-09 13-30-07

**Comment by SidharthBansal** [comment]
I don't think so. Please check by doing
``git log -n 2``
paste the log here

**Comment by dinaelhanan** [comment]
@SidharthBansal :
commit d4ccf6b6d061b2d898a2eb607d0dd2b045856081 (HEAD -> master, origin/master)
Author: Sidharth Bansal <bansal.sidharth2996@gmail.com>
Date:   Fri Dec 7 00:14:17 2018 +0530

    OAuth shown in development mode (#4148)

commit b5d889eb0737218b6348f5fc5d7ad30a9d57c1f3
Author: Oorjit Chowdhary <39333058+oorjitchowdhary@users.noreply.github.com>
Date:   Thu Dec 6 20:34:43 2018 +0530

    Spam button for /comments (#4143)

**Comment by SidharthBansal** [comment]
https://github.com/publiclab/plots2/commits/master
check this out.
You need to reset your master to current status of master by 
``git fetch upstream`` and then
``git reset --hard upstream/master``

**Comment by dinaelhanan** [comment]
@SidharthBansal I am getting this for git fetch upstream:
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
and for git reset --hard upstream/master I get:
fatal: ambiguous argument 'upstream/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'

**Comment by dinaelhanan** [comment]
@JonathanXu1 thank you!!!

**Comment by dinaelhanan** [comment]
@JonathanXu1 i did all the above steps and everything is up to date but I am still not getting the modal when clicking login or signup from the main page

**Comment by kevinzluo** [comment]
@dinaelhanan I just checked your branch [here](https://github.com/dinaelhanan/plots2/tree/linkloginmodal) and it says your branch is 4 commits ahead and 13 commits behind publiclab:master. This means that it is still not up to date.

Can you type `git remote -v` and see if an upstream is listed? Please send the output here.

**Comment by JonathanXu1** [comment]
Okay so you now have the upstream remote
Now pull from the upstream to your master branch, and rebase your current branch
https://publiclab.org/wiki/contributing-to-public-lab-software#Rewinding+the+master+branch

**Comment by ValentinaTironi** [comment]
I did this https://stackoverflow.com/questions/7244321/how-do-i-update-a-github-forked-repository
Works for me!

**Comment by dinaelhanan** [comment]
@JonathanXu1 thanks for that, it worked! Wonderful modal by the way! @kevinzluo @ValentinaTironi I didn't see your suggestions, but regardless, thanks for the help!

**Comment by dinaelhanan** [comment]
@SidharthBansal OK I think everything is good now :)

**Comment by SidharthBansal** [comment]
gif please.

File: test/fixtures/node_tags.yml (hunk 1 above)
Comment (SidharthBansal) [reply test/fixtures/node_tags.yml]: Please remove these changes.

Comment (SidharthBansal) [reply test/fixtures/node_tags.yml]: Don't resolve the suggested changes.

**Comment by dinaelhanan** [comment]
@SidharthBansal I don't know how to remove those changes in the node tag file though

File: test/fixtures/node_tags.yml (hunk 1 above)
Comment (SidharthBansal) [reply test/fixtures/node_tags.yml]: I mean press the resolve button. You need to untrack this file from the last commit.

**Comment by SidharthBansal** [comment]
https://stackoverflow.com/questions/12481639/remove-files-from-git-commit
this might help you

**Comment by SidharthBansal** [comment]
Please provide the complete gif. 
What happens on successful sign up and what happens on unsuccessful sign up?

**Comment by SidharthBansal** [comment]
Follow these sequences
join now clicked -> login modal -> successful sign up -> should be present on the /dashboard
second sequence
join now clicked -> login modal -> unsucessful sign up -> /login -> fill correct data ->redirected to /dashboard

**Comment by dinaelhanan** [comment]
sorry for the delay! 
join now clicked -> login modal -> successful sign up -> should be present on the /dashboard
https://drive.google.com/file/d/1hVgH_VLRjEIqifd8fJEBZVOpLW_YSI0m/view?usp=sharing
 join now clicked -> login modal -> unsucessful sign up -> /login -> fill correct data ->redirected to /dashboard (note that I saved an incorrect password to use so that file size can be small)
https://drive.google.com/file/d/1tookJr7b6d6L-KR-OFMFc2SThTNsFvwb/view?usp=sharing

**Comment by dinaelhanan** [comment]
@gauravano I've also completed this task. GCI claiming of tasks has ended and I can't get that credit for this, but still! Enjoyed doing it :smile:

**Pull request merged**